jobs/healing/pipeline-healing.js

```diff
@@ -229,23 +232,91 @@ async function getScopedSourceTriples( config, property, conceptSchemeUri, publi
   return reformatQueryResult(result);
 }
 
+function arrayToFile(array, file){
+  let fd = fs.openSync(file, "w");
+  for (let i=0; i<array.length; ++i){
+    fs.writeSync(fd, JSON.stringify(array[i]) + "\n")
+  }
+  fs.close(fd)
+  return file
+}
+function getTempFile() {
+  let file = "/tmp/healing-" + uuid();
+  return file;
+}
+function lines(filename) {
+  let retval = []
+  let rl = new readlines(filename)
+  let line
+  while ((line = rl.next())) {
+    line = line.toString()
+    if (line[0] === "<" || line[0] === ">") line = line.replace(/^./, "");
+    retval.push(JSON.parse(line) || {})
```

Comment:
neither the json parse nor the || 

Comments might be welcome :-)

Reply:
the json parse is to convert the line back to an Object, the || was put there to avoid parse errors stopping the entire pipeline.
I will add a comment for it!

Reply:
but not sure how line can be valid json string, and how || will catch parse exceptions? I miss somthing.

Reply:
line is a valid json string because the dataset gets converted into lines by the `arrayToFile` function which makes each element a single line (using JSON.stringify) which means here we parse such line which comes from the stringify function